MekHQ/src/mekhq/gui/CampaignGUI.java

```diff
@@ -1643,10 +1643,16 @@ private void btnGMModeActionPerformed(java.awt.event.ActionEvent evt) {// GEN-FI
 
     private void menuOptionsActionPerformed(java.awt.event.ActionEvent evt) {// GEN-FIRST:event_menuOptionsActionPerformed
         boolean atb = getCampaign().getCampaignOptions().getUseAtB();
+        boolean timein = getCampaign().getCampaignOptions().getUseTimeInService();
         boolean staticRATs = getCampaign().getCampaignOptions().useStaticRATs();
         CampaignOptionsDialog cod = new CampaignOptionsDialog(getFrame(), true,
                 getCampaign(), getIconPackage().getCamos());
         cod.setVisible(true);
+        if (timein != getCampaign().getCampaignOptions().getUseTimeInService()) {
```

Comment:
Does this do anything?  I read this as `if (getCampaign().getCampaignOptions().getUseTimeInService() != getCampaign().getCampaignOptions().getUseTimeInService())`.  You set `timein` to a value, and then 5 lines later check to see if the value is still the same.  I feel like this should just be:
```
if (timein) {
    getCampaign().initTimeInService();
}
```

Reply:
That was a straight rewrite of the atb init so never paid close attention. Will change and test.

Reply:
I am not sure why, but if I make the requested changes it causes issues. As is everything works fine, after the changes the first time the I check the option nothing happens. If I uncheck the option, save options, then recheck the option it works right.